File: LeetCode/129. Sum Root to Leaf Numbers/129.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        
        return Srln(root,0);
    }
    public int Srln(TreeNode node,int currentNumber){
        if(node==null){
            return 0;
        }
        currentNumber=currentNumber*10+node.val;
        if(node.left==null&&node.right==null){
            return currentNumber;
        }
        return Srln(node.left, currentNumber) + Srln(node.right, currentNumber);
    }
}

File: LeetCode/129. Sum Root to Leaf Numbers/129.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def S(node,currentNumber):
            if not node:
                return 0
            currentNumber=currentNumber*10+node.val
            if not node.left and not node.right:
                return currentNumber
            return S(node.left,currentNumber)+S(node.right,currentNumber)
        return S(root,0)